### PR TITLE
bug 1082798

### PR DESCRIPTION
**Pull request by m8ttyB**

### CODE REVIEW

**Comment by groovecoder** [comment]
Ping @darkwing - what do you think about this? Merge?

**Comment by darkwing** [comment]
If it blocks merging of PRs, I'd say no, since we noticed some inconsistencies in tests passing locally vs. on Sauce Labs.

File: .travis.yml [hunk 1]
```diff
@@ -32,12 +39,18 @@ install:
...
+    - sudo apt-get install -y nodejs
 
 before_script:
     - mysql -e 'create database kuma;'
 
 script:
     - coverage run --include=`pwd`/kuma/*,`pwd`/apps/* manage.py test --noinput -v2 access actioncounters authkeys contentflagging devmo kpi kuma landing sumo
+    - cd tests/ui
+    - npm install intern
+    - node_modules/.bin/intern-runner config=intern-sauce-labs d=developer.allizom.org
```
Comment (stof) [inline .travis.yml]: ``if [ "$TRAVIS_SECURE_VARS" = 1 ]; then node_modules/.bin/intern-runner config=intern-sauce-labs d=developer.allizom.org; fi;`` will allow running these tests only when secure vars are available on Travis

File: .travis.yml [hunk 2]
```diff
@@ -32,12 +39,18 @@ install:
...
+    - curl -sL https://deb.nodesource.com/setup | sudo bash -
+    - sudo apt-get install -y nodejs
 
 before_script:
     - mysql -e 'create database kuma;'
 
 script:
     - coverage run --include=`pwd`/kuma/*,`pwd`/apps/* manage.py test --noinput -v2 access actioncounters authkeys contentflagging devmo kpi kuma landing sumo
+    - cd tests/ui
+    - npm install intern
```
Comment (stof) [inline .travis.yml]: the npm install should not be part of the testing scripts but of the setup (probably ``install`` in that case). A failure during one of the scripts does not stop Travis from running other ones (this is a main difference with other steps of the build)

File: .travis.yml [hunk 3]
```diff
@@ -32,12 +39,18 @@ install:
     - pip install --no-index --find-links=travis_wheels -r requirements/compiled.txt
     - pip install --find-links=base_wheels -r requirements/prod.txt -r requirements/dev.txt
     - pip install coveralls
+    # JS test requirements
+    - curl -sL https://deb.nodesource.com/setup | sudo bash -
+    - sudo apt-get install -y nodejs
```
Comment (stof) [inline .travis.yml]: nodejs is already available on Travis (for the python language, you don't have the choice of the nodejs version, but you still have it available)

File: .travis.yml (hunk 1 above)
Comment (m8ttyB) [inline .travis.yml]: thank you .. done

File: .travis.yml (hunk 2 above)
Comment (m8ttyB) [inline .travis.yml]: Good point -- done

File: .travis.yml (hunk 3 above)
Comment (m8ttyB) [inline .travis.yml]: That's good to know -- I'll remove the installation of nodejs from the `.travis.yaml` config and see if presents a problem

**Comment by darkwing** [comment]
Continued pause on this until Q2; going to close for now but we'll come back to it @m8ttyB !

**Pull request closed without merging**